bigchaindb/common/schema/transaction.yaml

```diff
@@ -0,0 +1,248 @@
+    additionalProperties: false
+    required:
+    - id
+    properties:
+      id:
+        "$ref": "#/definitions/uuid4"
+      divisible:
+        type: boolean
+        description: |
+            Whether or not the asset has a quantity that may be partially spent.
+      updatable:
+        type: boolean
+        description: |
+            Whether or not the description of the asset may be updated. Defaults to false.
+      refillable:
+        type: boolean
+        description: |
+          Whether the amount of the asset can change after its creation. Defaults to false.
+      data:
+        description: |
+          User provided metadata associated with the asset. May also be NULL.
+        anyOf:
+        - type: object
+          additionalProperties: true
+        - type: 'null'
+  condition:
+    type: object
+    description: |
+        An output of a transaction. A condition describes a quantity of an asset
+        and what conditions must be met in order for it to be fulfilled. See also:
+        fulfillment_.
+    additionalProperties: false
+    required:
+    - owners_after
+    - condition
+    - amount
+    properties:
+      cid:
+        type: integer
+        description: |
+            Index of this transaction's appearance in the `transaction.conditions`_
+            array. In a transaction with 2 conditions, the `cid`s will be 0 and 1.
+      condition:
+        description: |
+            Body of the condition. Has the properties:
+
+            - **details**: Details of the condition.
+            - **uri**: Condition encoded as an ASCII string.
+        type: object
+        additionalProperties: false
+        required:
+        - uri
+        properties:
+          details:
+            type: object
+            additionalProperties: true
+          uri:
+            type: string
+      owners_after:
+        "$ref": "#/definitions/owners_list"
+        description: |
+            List of public keys associated with asset ownership at the time
+            of the transaction.
+      amount:
+        type: integer
+        description: |
+            Integral amount of the asset represented by this condition.
+            In the case of a non divisible asset, this will always be 1.
+  fulfillment:
+    type: "object"
+    description:
+      A fulfillment is an input to a transaction, named as such because it
+      fulfills a condition of a previous transaction. In the case of a
+      **CREATE** transaction, a fulfillment may provide no `input`.
+    additionalProperties: false
+    required:
+    - owners_before
+    - input
+    - fulfillment
+    properties:
+      fid:
+        type: integer
+        description: |
+            The offset of the fulfillment within the fulfillents array.
+      owners_before:
+        "$ref": "#/definitions/owners_list"
+        description: |
+            List of public keys of the previous owners of the asset.
+      fulfillment:
+        anyOf:
+        - type: object
```

Comment:
Mmm, we can keep the object version of `fulfillment` here, but, depending on when we expect to do validation, this might not be necessary. The object version should only be used when fulfillments haven't been signed yet, so this is strictly something only the driver would want.